src/extension/infer.rs

```diff
@@ -302,6 +302,18 @@ impl UnificationContext {
                 }
             }
 
+            if node_type.tag() == OpTag::Cfg {
+                let mut children = hugr.children(node);
+                let entry = children.next().unwrap();
+                let exit = children.next().unwrap();
+                let m_entry = self.make_or_get_meta(entry, Direction::Incoming);
+                let m_exit_in = self.make_or_get_meta(exit, Direction::Incoming);
+                let m_exit_out = self.make_or_get_meta(exit, Direction::Outgoing);
+                self.add_constraint(m_input, Constraint::Equal(m_entry));
```

Comment:
It seems a bit odd that you deal with both Direction::Incoming and Direction::Outgoing for the exit node, but not for the entry node. Do you mean to do similarly for `self.make_or_get_meta(entry, Direction::Outgoing)` too?

If so, how about commoning up with the case for nested dataflow nodes above, i.e. breaking this block out into a helper function:
```
for dir in Direction::BOTH {
                    let m_input_node = self.make_or_get_meta(input, dir);
                    self.add_constraint(m_input_node, Constraint::Equal(m_input));
                    let m_output_node = self.make_or_get_meta(output, dir);
                    self.add_constraint(m_output_node, Constraint::Equal(m_output));
                }
```

Reply:
The exit node of a CFG can never add extension requirements between its inputs and outputs, but an entry node is an arbitrary DFB with an extension_delta, so the output extensions aren't necessarily the same as the CFG's input extensions.

Having said that, we _can_ add a constraint that the entry node's outputs are the CFG input plus that node's delta

Reply:
Ah yes sorry, I forget that Entry could have anything in it. Hmmm. So that case is handled elsewhere. Does the op_signature for an Output node not specify that it adds no delta, then? (i.e. do we need to make a constraint for Exit node output extensions here at all?)

Reply:
Yes you're right the output constraint is probably redundant

Reply:
Ok in that case I suggest you do either *one* of the following two options:
* Don't add output constraint for either node. *Ideally* I think you'd make the same change to the `if let Some([input, output]) = hugr.get_io(node)` block above too, and perhaps elsewhere as a general principle, tho, and that might be better done in a later PR. In which case:
* Just add a comment here as to why entry and exit are different. (I should know, I wrote the spec for them, but having not worked on CFGs for awhile I forgot, and so others will too - it is an oddity of the spec that they are not symmetric, so I think that justifies a comment.)